core/objects.go

```diff
@@ -254,6 +274,10 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord *ValidationRecord `json:"validationRecord,omitempty"`
```

Comment:
The second is clearly better.  It means that a generic processor can get as far as the `{stuff}` object without knowing anything about the challenge type itself.